app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/Filter.php

```diff
@@ -58,22 +53,19 @@ class Filter implements ProductQueryInterface
     /**
      * @param SearchResultFactory $searchResultFactory
      * @param ProductProvider $productDataProvider
-     * @param LayerResolver $layerResolver
      * @param FieldSelection $fieldSelection
      * @param SearchCriteriaBuilder $searchCriteriaBuilder
      * @param ScopeConfigInterface $scopeConfig
      */
     public function __construct(
         SearchResultFactory $searchResultFactory,
         ProductProvider $productDataProvider,
-        LayerResolver $layerResolver,
```

Comment:
If the dependency is not used at all, it should be ok to remove it. In GraphQl area, we only treat schema as out public api, so we can make changes like this to code that is not marked `@api`

Reply:
It's not so bad to remove it actually
Do this change and we'll get it approved. Since this implements an interface, the class was not meant to be extended anyway.